### PR TITLE
move code which clears the pos and alt hold flags to the correct place.

### PR DESCRIPTION
**Pull request by tumbili**
this should fix the problem of the jump in the altitude setpoint when
switching into altitude hold mode.

### CODE REVIEW

**Comment by LorenzMeier** [comment]
@tumbili Can you rebase on master to fix CI? How do we get this tested?

**Comment by tumbili** [comment]
@LorenzMeier I already rebased and it didn't work. I can try again.

**Comment by tumbili** [comment]
@LorenzMeier Tomorrow is testing day. I have planned to take the 450 with me, then I can also test optical flow outdoors.

**Comment by ecmnet** [comment]
Did some flight tests with that change. The jump still occurs :
http://logs.uaventure.com/view/uC2rDHaUAbbgF98NMnpGPZ#Local_Z_PLOT

**Comment by ecmnet** [comment]
Since leaving out the FF term when resetting the set point again solved the issue =>

       _pos_sp(2) = _pos(2);   // + (_vel(2) - _params.vel_ff(2) * _vel_sp(2)) / _params.pos_p(2);

the problem cannot be the point of time when *reset_alt_sp()* is called.

**Comment by AndreasAntener** [comment]
@ecmnet could you tell us how you were able to trigger the issue? Seems to be tricky to reproduce..

**Comment by ecmnet** [comment]
First of all: I use https://github.com/PX4/Firmware/pull/3532 for position estimation based on LIDAR without any GPS. I also used a very low Baro weight (as I am testing indoors with an 450 vehicle). But I agree, it is very hard to reproduce as it does occurs only in 30-50% of the cases. Basically I do a manual start to about 0.5 to 1m, then switch to altitude hold mode. Thats all I did. Although the fix of Roman seems to make sense, it is also obvious, that *reset_alt_sp()* is called at the correct point of time. This is because my quick fix solved the issue for me. And it would not, if *reset_alt_sp()* is not called in time.

**Comment by tumbili** [comment]
@ecmnet Ok, I have pushed another fix which resets position and altitude to the actual position of altitude. Before this was done differently, because the position controller always controlled position directly. Let me test this in sim.

**Comment by ecmnet** [comment]
@tumbili As for altitude hold, thats exactly what I did. But I have to admit, I do not clearly understand the fix. Would you mind to explain a bit?

**Comment by ecmnet** [comment]
AltHold flight tested. Works.

File: src/modules/mc_pos_control/mc_pos_control_main.cpp [hunk 1]
```diff
@@ -677,12 +677,8 @@ MulticopterPositionControl::reset_pos_sp()
 	if (_reset_pos_sp) {
 		_reset_pos_sp = false;
 		/* shift position setpoint to make attitude setpoint continuous */
-		_pos_sp(0) = _pos(0) + (_vel(0) - PX4_R(_att_sp.R_body, 0, 2) * _att_sp.thrust / _params.vel_p(0)
```
Comment (LorenzMeier) [inline src/modules/mc_pos_control/mc_pos_control_main.cpp]: Why did this get removed? Did you carefully check the rate outputs before and after?

**Comment by LorenzMeier** [comment]
@tumbili Have you ensured that the system now doesn't twitch on forward flight when switching modes? The changes look like they remove input handling code we want, and I don't see how its handled.

**Comment by LorenzMeier** [comment]
@tumbili Could you please answer my questions on this one? I'd like to get this in. A fix should not dangle for 3 weeks.

**Comment by tumbili** [comment]
@LorenzMeier I'm looking at it. We cannot use the same reset mechanism for the positions anymore since we have party moved to direct velocity control.

**Comment by LorenzMeier** [comment]
@tumbili Sounds like this here is then for tomorrow.

**Comment by tumbili** [comment]
@LorenzMeier Yes, I have done tests for this today, will push something tomorrow.

**Comment by tumbili** [comment]
Replaced by #3822

**Pull request closed without merging**